File: app/src/main/java/com/lidboud/kotlin/data/model/domain/Resource.kt
package com.lidboud.kotlin.data.model.domain

import com.google.gson.annotations.SerializedName

data class Resource(@field:SerializedName("_id") val id: String = "",
                    @field:SerializedName("title") val title: String = "",
                    @field:SerializedName("description") val description: String = "",
                    @field:SerializedName("addresses") val addresses: List<Address> = listOf(),
                    @field:SerializedName("contactInfo") val contactInfo: ContactInfo)

data class Address(@field:SerializedName("address1") val address: String = "",
                   @field:SerializedName("city") val city: String = "",
                   @field:SerializedName("zipCode") val zipCode: String = "",
                   @field:SerializedName("country") val country: String = "",
                   @field:SerializedName("gps") val gps: Gps? = null)


data class Gps(@field:SerializedName("latitude") val latitude: String,
               @field:SerializedName("longitude") val longitude: String)

class ContactInfo(@field:SerializedName("email") val email: List<String>?,
                  @field:SerializedName("website") val website: List<String>?)

File: mobile-test/mobile-test/Model/Resource.swift
//
//  Restaurant.swift
//  mobile-test
//
//  Created by Ali Pishvaee on 4/25/19.
//  Copyright © 2019 Ali Pishvaee. All rights reserved.
//

import Foundation
import UIKit

struct Resource: Codable {
    var id: String
    var slug: String
    var eid: String
    var title: String
    var description: String
    var categoryEid: String
    var v: Int
    var photo: String
    var active: Bool
    var updatedAt: String
    var createdAt: String
    var addresses: [Address]?
    var freeText: [Address]?
    var contactInfo: ContactInfo
    var bizHours: BizHours?
    var socialMedia: SocialMedia?
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case slug, eid, title, description
        case categoryEid = "category_eid"
        case v = "__v"
        case photo
        case active = "_active"
        case updatedAt = "updated_at"
        case createdAt = "created_at"
        case addresses, contactInfo, bizHours, socialMedia, freeText
    }
}




struct Address: Codable {
    let address1: String?
    var label: String?
    var zipCode: String?
    var city: String?
    let state: String?
    var country: String?
    let gps: Gps?
}

struct Gps: Codable {
    var latitude: String
    var longitude: String
}

struct BizHours: Codable {
    var sunday: Nday?
    var monday: Nday?
    var tuesday: Nday?
    var wednesday: Nday?
    var thursday: Nday?
    var friday: Nday?
    var saturday: Nday?
}

struct Nday: Codable {
    var from: String
    var to: String
}

struct ContactInfo: Codable {
    var website: [String]?
    var email: [String]?
    var faxNumber: [String]?
    var tollFree: [String]?
    var phoneNumber: [String]
}

struct SocialMedia: Codable {
    var youtubeChannel: [String]
    var twitter: [String]
    var facebook: [String]
}



extension Array where Element == Resource {
    func matching(_ text: String?) -> [Resource] {
        if let text = text, text.count > 0 {
            return self.filter {
                $0.title.contains(text)
                    || $0.description.contains(text)
            }
        } else {
            return self
        }
    }
    
}
